PusherServer.Tests/UnitTests/AuthenticateUser.cs

```diff
@@ -0,0 +1,129 @@
+
+            UserData userData = new UserData()
+            {
+                id = "unique_user_id",
+            };
+            string userDataJson = DefaultSerializer.Default.Serialize(userData);
+
+            string expectedAuthString = Config.AppKey + ":" + CreateSignedString(socketId, userDataJson);
+
+            IUserAuthenticationResponse result = _pusher.AuthenticateUser(socketId, userData);
+            Assert.AreEqual(expectedAuthString, result.auth);
+            Assert.AreEqual(userDataJson, result.user_data);
+        }
+
+        [Test]
+        public void with_watchlist_the_auth_response_is_valid()
+        {
+            string socketId = "123.456";
+
+            UserData userData = new UserData()
+            {
+                id = "unique_user_id",
+                watchlist = new string[] { "user1", "user2" },
+                user_info = new { twitter_id = "@leggetter" }
+            };
+            string userDataJson = DefaultSerializer.Default.Serialize(userData);
+
+            string expectedAuthString = Config.AppKey + ":" + CreateSignedString(socketId, userDataJson);
+
+            IUserAuthenticationResponse result = _pusher.AuthenticateUser(socketId, userData);
+            Assert.AreEqual(expectedAuthString, result.auth);
+            Assert.AreEqual(userDataJson, result.user_data);
+        }
+
+        [Test]
+        public void with_userinfo_the_auth_response_is_valid()
+        {
+            string socketId = "123.456";
+
+            UserData userData = new UserData()
+            {
+                id = "unique_user_id",
+                user_info = new { twitter_id = "@leggetter" }
+            };
+            string userDataJson = DefaultSerializer.Default.Serialize(userData);
+
+            string expectedAuthString = Config.AppKey + ":" + CreateSignedString(socketId, userDataJson);
+
+            IUserAuthenticationResponse result = _pusher.AuthenticateUser(socketId, userData);
+            Assert.AreEqual(expectedAuthString, result.auth);
+            Assert.AreEqual(userDataJson, result.user_data);
+        }
+
+        [Test]
+        public void with_userinfo_and_watchlist_the_auth_response_is_valid()
+        {
+            string socketId = "123.456";
+
+            UserData userData = new UserData()
+            {
+                id = "unique_user_id",
+                watchlist = new string[] { "user1", "user2" },
+                user_info = new { twitter_id = "@leggetter" }
+            };
+            string userDataJson = DefaultSerializer.Default.Serialize(userData);
+
+            string expectedAuthString = Config.AppKey + ":" + CreateSignedString(socketId, userDataJson);
+
+            IUserAuthenticationResponse result = _pusher.AuthenticateUser(socketId, userData);
+            Assert.AreEqual(expectedAuthString, result.auth);
+            Assert.AreEqual(userDataJson, result.user_data);
+        }
+
+        [Test]
+        public void channel_data_is_encoded_as_JSON()
```

Comment:
redundant test case here

Reply:
I agree it is redundant. I'll remove this one.